paasta_tools/cli/cmds/spark_run.py

```diff
@@ -154,47 +160,76 @@ def get_docker_run_cmd(
     for volume in volumes:
         cmd.append('--volume=%s' % volume)
     cmd.append('%s' % docker_img)
-    cmd.append(docker_cmd)
+    cmd.extend(('sh', '-c', docker_cmd))
 
     return cmd
 
 
-def get_spark_configuration(
+def get_spark_env(
+    cmd,
+    spark_conf,
+):
+    spark_env = {}
+
+    creds = Session().get_credentials()
+    spark_env['AWS_ACCESS_KEY_ID'] = creds.access_key
+    spark_env['AWS_SECRET_ACCESS_KEY'] = creds.secret_key
+
+    if cmd == 'jupyter':
```

Comment:
You probably need

    if 'jupyter' in cmd:

if you want this to be tolerant of many forms of running a notebook, no?

Why not just set these regardless though?

Reply:
At this point, we launch the jupyter notebook server with the options that we pick (e.g. directories etc). There will be conflicts if we let users start jupyter with arbitrary options.